hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/DatabaseConfiguration.java

```diff
@@ -1,15 +1,85 @@
 package org.hibernate.reactive.containers;
 
+import java.util.Arrays;
+
 /**
  * Contains the common constants that we need for the configuration of databases
  * during tests.
  */
-public interface DatabaseConfiguration {
-
-	boolean USE_DOCKER = Boolean.getBoolean("docker");
+public class DatabaseConfiguration {
 
-	String USERNAME = "hreact";
-	String PASSWORD = "hreact";
-	String DB_NAME = "hreact";
+	public static final boolean USE_DOCKER = Boolean.getBoolean("docker");
+	
+	public static enum DBType {
+		DB2,
+		MYSQL,
+		POSTGRESQL
+	}
+	
+	public static final String USERNAME = "hreact";
+	public static final String PASSWORD = "hreact";
+	public static final String DB_NAME = "hreact";
+	
+	private static DBType dbType;
+	
+	public static DBType dbType() {
+		if (dbType == null) {
+			String dbTypeString = System.getProperty( "db", DBType.POSTGRESQL.name() ).toUpperCase();
+			if ("PG".equals( dbTypeString ) || "POSTGRES".equals( dbTypeString ))
```

Comment:
If I use:
`gradle clean build -Pdb=pg`
instead of
`gradle clean build -Pdb=POSTGRESQL`
only one test fails:

```
Gradle Test Executor 35 > org.hibernate.reactive.LazyManyToOneAssociationTest > fetchWithOneAuthor FAILED              
    java.lang.IllegalArgumentException: Unknown DB type 'PG' specified. Allowed values are: [DB2, MYSQL, POSTGRESQL]   
        at org.hibernate.reactive.containers.DatabaseConfiguration.dbType(DatabaseConfiguration.java:34)                                                                                                                                      
        at org.hibernate.reactive.containers.DatabaseConfiguration.getJdbcUrl(DatabaseConfiguration.java:42)           
        at org.hibernate.reactive.BaseReactiveTest.constructConfiguration(BaseReactiveTest.java:56)                                                                                                                                           
        at org.hibernate.reactive.LazyManyToOneAssociationTest.constructConfiguration(LazyManyToOneAssociationTest.java:16)                                                                                                                   
                                                                                                                                                                                                                                              
        Caused by:                                                                                                                                                                                                                            
        java.lang.IllegalArgumentException: No enum constant org.hibernate.reactive.containers.DatabaseConfiguration.DBType.PG                                                                                                                
            at java.base/java.lang.Enum.valueOf(Enum.java:240)                                                         
            at org.hibernate.reactive.containers.DatabaseConfiguration$DBType.valueOf(DatabaseConfiguration.java:13)                                                                                                                          
            at org.hibernate.reactive.containers.DatabaseConfiguration.dbType(DatabaseConfiguration.java:31)           
            ... 3 more                                                                                                 
                                                                                                                                                                                                                                              
    java.lang.NullPointerException                                                                                     
        at org.hibernate.reactive.BaseReactiveTest.after(BaseReactiveTest.java:87)                                                                                                                                                            
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                                                                                                                                     
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                                                                                                                                   
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)            
```

Reply:
Hmm, that's a bug. Let me look into that.

Reply:
Fixed and updated